diff --git a/dist/index.cjs b/dist/index.cjs
index 52f4b7d7aaccb3bd2566ad9b6c55de54fef25e81..eb5f78a16c3c1902bc24ba76a83d7a837d079b73 100644
--- a/dist/index.cjs
+++ b/dist/index.cjs
@@ -42,7 +42,7 @@ var Result = class Result {
 	static try(result, ...args) {
 		try {
 			if (typeof result === "function") result = result.apply(void 0, args);
-			if (result instanceof Promise) return result.then(Result.ok, Result.error);
+			if (result != null && typeof result.then === "function") return Promise.resolve(result).then(Result.ok, Result.error);
 			return Result.ok(result);
 		} catch (error$1) {
 			return Result.error(error$1);
diff --git a/lib/index.js b/lib/index.js
index 33a6e0c80c9af12c98d7447055592ba7e1b7b191..cf47208b1ce4ace9c1c664be3882e64229a6c59b 100644
--- a/lib/index.js
+++ b/lib/index.js
@@ -58,9 +58,9 @@ export class Result {
         result = result.apply(undefined, args);
       }
 
-      // Promises must return a valid Promise<Result<T>>
-      if (result instanceof Promise) {
-        return result.then(Result.ok, Result.error);
+      // Promises and thenables must return a valid Promise<Result<T>>
+      if (result != null && typeof result.then === 'function') {
+        return Promise.resolve(result).then(Result.ok, Result.error);
       }
 
       // If the result is not a function or a Promise, we can be sure its a success
